### PR TITLE
Backend feature/order

### PR DESCRIPTION
**Pull request by burakberk**
Implemented order.

Endpoint:

customer/order -> parameters: _id

Example return:

   ```
 [{
        "productId": "5fe8a60c0780902a50c955b8",
        "vendorId": "5fc691e76666a7596672ce38",
        "amount": 7,
        "price": 420,
        "shipmentPrice": 15,
        "cargoCompany": "PTT",
        "enoughLeft": true
    }, ...]
```

Transforms all items in the shoppingCart into an order array. For each item, fetches their current price, shipmentPrice, cargoCompany and checks whether there is enough amount left to buy.

Example requests can be accessed from this [postman collection.](https://www.getpostman.com/collections/4fb935cf830feb85e94a)

### CODE REVIEW

File: backend/services/order.js [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+      temp["cargoCompany"] = current_product["vendorSpecifics"][0]["cargoCompany"];
+      if (current_product["vendorSpecifics"][0]["amountLeft"] > current["amount"]) {
+        temp["enoughLeft"] = true;
+      } else {
+        temp["enoughLeft"] = false;
+      }
+      items.push(temp);
+    }
+  }
+  // console.log(items);
```
Comment (oztasozgurcan) [inline backend/services/order.js:41]: remove unnecessary comments!

File: backend/controllers/order.js [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+const AppValidator = require("../util/appValidator");
+const BaseUtil = require("../util/baseUtil");
+const Messages = require("../util/messages");
+const BB = require("bluebird");
+const factory = require("../services/crudFactory");
+const Customer = require("../models/customer");
+const OrderService = require("../services/order");
+
+exports.createOrderController = BaseUtil.createController((req) => {
+  let { _id } = req.query;
```
Comment (oztasoi) [inline backend/controllers/order.js]: we use body rather than query, so it should be changed into body

File: backend/services/order.js [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+  // console.log(shoppingCart['shoppingCart'][0]);
```
Comment (oztasoi) [inline backend/services/order.js]: remove unnecessary comments!

**Review by oztasoi — CHANGES_REQUESTED** [review]

**Comment by burakberk** [comment]
PR updated according to change requests.

**Review by oztasoi — APPROVED** [review]
Seems fine, let's see how it works in development environment 🚢

**Pull request merged**